code/ai/ai_profiles.cpp

```diff
@@ -486,20 +475,14 @@ void parse_ai_profiles_tbl(const char *filename)
 					mprintf(("Warning: \"$perform less checks for death screams\" flag is deprecated in favor of \"$perform fewer checks for death screams\"\n"));
 					bool temp;
 					stuff_boolean(&temp);
-					if (temp)
-						profile->flags2 |= AIPF2_PERFORM_FEWER_SCREAM_CHECKS;
-					else
-						profile->flags2 &= ~AIPF2_PERFORM_FEWER_SCREAM_CHECKS;
+                    profile->flags.set(AI::Profile_Flags::Perform_fewer_scream_checks, temp);
 				}
 				if (optional_string("$allow primary link delay:"))
 				{
 					mprintf(("Warning: \"$allow primary link delay\" flag is deprecated in favor of \"$allow primary link at mission start\"\n"));
 					bool temp;
 					stuff_boolean(&temp);
-					if (temp)
-						profile->flags2 &= ~AIPF2_ALLOW_PRIMARY_LINK_AT_START;
-					else
-						profile->flags2 |= AIPF2_ALLOW_PRIMARY_LINK_AT_START;
+                    profile->flags.set(AI::Profile_Flags::Allow_primary_link_at_start, temp);
```

Comment:
I think this needs to be inverted. The previous version disabled the flag then `temp` was true, now it enables it.